File: golang/src/abac.go
package main

import (
	"fmt"

	"github.com/casbin/casbin/v2"
)

func abac() {
	e, _ := casbin.NewEnforcer("../config/model.conf", "../config/policy.csv")

	// Print the policy to log.
	e.EnableLog(true)

	// Check the permission.
	sub := "alice" // the user that wants to access a resource.
	obj := "data2" // the resource that is going to be accessed.
	act := "read"  // the operation that the user performs on the resource.

	res, err := e.Enforce(sub, obj, act)
	if err != nil {
		panic(err)
	}

	if res {
		// permit alice to read data1
		fmt.Println("permit alice to read data1")
	} else {
		// deny the request, show an error
		fmt.Println("deny the request, show an error")
	}
}

File: nodejs/script/abac.js
//import { newEnforcer } from 'casbin';
const { newEnforcer } = require('casbin');

async function main() {
const e =  await newEnforcer('../config/model.conf', '../config/policy.csv');

const sub = 'alice'; // 想要访问资源的用户
const obj = 'data2'; // 将要被访问的资源
const act = 'read'; // 用户对资源进行的操作

const res = await e.enforce(sub, obj, act);
if (res) {
    // 允许alice读取data1
    console.log('permit alice to read data1');
} else {
    // 拒绝请求，抛出异常
    console.log('deny the request');
}
}

main();
